verification/cbmc/proofs/aws_cryptosdk_priv_hdr_parse_header_version/aws_cryptosdk_priv_hdr_parse_header_version_harness.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/cryptosdk/header.h>
+#include <aws/cryptosdk/private/header.h>
+#include <make_common_data_structures.h>
+#include <proof_helpers/make_common_data_structures.h>
+#include <proof_helpers/proof_allocators.h>
+#include <proof_helpers/utils.h>
+
+void aws_cryptosdk_priv_hdr_parse_header_version_harness() {
+    /* Nondet Input */
+    struct aws_cryptosdk_hdr *hdr   = ensure_nondet_hdr_has_allocated_members(MAX_TABLE_SIZE);
+    struct aws_byte_cursor *pcursor = malloc(sizeof(*pcursor));
+    uint8_t *header_version         = malloc(sizeof(*header_version));
+
+    /* Assumptions */
+    __CPROVER_assume(aws_cryptosdk_hdr_members_are_bounded(hdr, MAX_EDK_LIST_ITEMS, MAX_BUFFER_SIZE));
+
+    /* Precondition: The edk list has allocated list elements */
+    ensure_cryptosdk_edk_list_has_allocated_list_elements(&hdr->edk_list);
+    __CPROVER_assume(aws_cryptosdk_hdr_is_valid(hdr));
+
+    __CPROVER_assume(hdr->enc_ctx.p_impl != NULL);
+    ensure_hash_table_has_valid_destroy_functions(&hdr->enc_ctx);
+    size_t empty_slot_idx;
+    __CPROVER_assume(aws_hash_table_has_an_empty_slot(&hdr->enc_ctx, &empty_slot_idx));
+
+    __CPROVER_assume(pcursor != NULL);
+    __CPROVER_assume(aws_byte_cursor_is_bounded(pcursor, MAX_BUFFER_SIZE));
+    ensure_byte_cursor_has_allocated_buffer_member(pcursor);
+    __CPROVER_assume(aws_byte_cursor_is_valid(pcursor));
+
+    __CPROVER_assume(header_version != NULL);
```

Comment:
Same as https://github.com/aws/aws-encryption-sdk-c/pull/644#discussion_r538953400 - can we pull these out into a helper function?

Reply:
Addressed with new hdr_setup function!